File: LeetCode/169. Majority Element/Solution.java
public class Solution {
    public int majorityElement(int[] nums) {
        int n = 0, count = 0;
        for (int num : nums) {
            if (count == 0) {
                n = num;
                count++;
            } else {
                if (n == num) {
                    count++;
                } else {
                    count--;
                }
            }
        }
        return n;
    }
}

File: LeetCode/169. Majority Element/solution.py
from collections import Counter


class Solution(object):
    def majorityElement(self, nums):
        """
        :type nums: List[int]
        :rtype: int
        """
        return sorted(nums)[len(nums) // 2]

    def majorityElement1(self, nums):
        c = Counter(nums)
        return c.most_common(1)[0][0]

    def majorityElement2(self, nums):
        t = n = 0
        for i in nums:
            if n == 0:
                t, n = i, 1
            elif t == i:
                n += 1
            else:
                n -= 1
        return t
